### PR TITLE
Fixes #12009 - Removed katello_test_runner and updated tests

### PR DESCRIPTION
**Pull request by johnpmitsch**
This removes katello test runner in preparation for the rails 4 conversion. in Minitest 5 (used by rails 4) [test runners are deprecated](http://blog.zenspider.com/releases/2013/05/minitest-version-5-0-0-has-been-released.html). The test runner was used to run a before_suite and after_suite function that was available to run before all tests. This functionality has been moved to the setup and teardown methods.

### CODE REVIEW

File: lib/katello/tasks/test.rake [hunk 1]
```diff
@@ -39,8 +49,82 @@ namespace :test do
...
+          ]
+          t.verbose = true
+        end
+
+        Rake::Task[test_task.name].invoke
+      end
+
+    end
+  end
+
```
Comment (johnpmitsch) [inline lib/katello/tasks/test.rake]: @ehelms I added these rake tasks for individual test folders we have been using for rails 4. They have proved to be useful and I figured others could benefit from using them, let me know if you want them removed.

File: test/glue/candlepin/consumer_test.rb [hunk 2]
```diff
@@ -6,43 +6,40 @@ module Katello
...
     include CandlepinConsumerSupport
 
-    @@dev = nil
-    @@org = nil
-    @@dev_cv = nil
-    @@dev_cve = nil
+    @dev = nil
+    @org = nil
+    @dev_cv = nil
+    @dev_cve = nil
```
Comment (ehelms) [inline test/glue/candlepin/consumer_test.rb]: These shouldn't be needed if switching to instance variables

File: lib/katello/tasks/test.rake (hunk 1 above)
Comment (ehelms) [inline lib/katello/tasks/test.rake]: I think they are fine to have.

**Comment by ehelms** [comment]
Mind giving this its own issue since it is going into master?

**Comment by johnpmitsch** [comment]
@ehelms ya np

**Comment by johnpmitsch** [comment]
[test]

**Comment by johnpmitsch** [comment]
[test]

**Comment by ehelms** [comment]
I am seeing failures when I run this in live mode.

**Comment by johnpmitsch** [comment]
[test]

**Comment by johnpmitsch** [comment]
[test]

**Comment by ehelms** [comment]
When this runs, I see a lot of the following in the logs:

```
(eval):4:in `must_raise'
/home/vagrant/katello/test/actions/katello/repository_set_test.rb:59:in `block in <class:EnableRepositoryTest>'
/home/vagrant/.rvm/gems/ruby-1.9.3-p448/gems/minitest-4.7.5/lib/minitest/unit.rb:1258:in `run'
```

**Comment by ehelms** [comment]
I also see this:

```
  1) Failure:
SourceCodeTest::best practices#test_0001_does not use ENV variables [/home/vagrant/katello/test/source_code_test.rb:46]:
Katello.config or Katello.early_config should be always used instead of ENV variables, Katello.config is
the single entry point to configuration. ENV variables are processed there.

check lines:
 - /home/vagrant/katello/test/katello_test_runner.rb:16: if ENV['suite']
 - /home/vagrant/katello/test/katello_test_runner.rb:18: suite.name == ENV['suite'].
Expected [["      if ENV['suite']\n", 16, "/home/vagrant/katello/test/katello_test_runner.rb"], ["          suite.name == ENV['suite']\n", 18, "/home/vagrant/katello/test/katello_test_runner.rb"]] to be empty.

```

File: test/source_code_test.rb [hunk 3]
```diff
@@ -80,7 +80,6 @@ def self.token_is_keyword?(str, lex, index, token)
               %r{engines/bastion/test/test_helper\.rb},
               %r{test/support/vcr\.rb},
               %r{test/support/runcible\.rb},
-              %r{test/katello_test_runner\.rb}, # TODO: clean up minitest_helper
```
Comment (johnpmitsch) [inline test/source_code_test.rb]: @ehelms for that particular failure you mentioned, it seems like this change was not there when you ran the test?

Comment (johnpmitsch) [inline test/source_code_test.rb]: I am getting passing tests locally

Comment (ehelms) [inline test/source_code_test.rb]: Yea - my checkout was in a weird state and seems fine now.

**Comment by ehelms** [comment]
Sounds like the logging of errors being raised will take further investigation, so I am good with this as is, thanks @johnpmitsch ! ACK

**Pull request merged**